### PR TITLE
feat: 🎸 Add validations to consumeAddMultiSigSignerAuth

### PR DESCRIPTION
**Pull request by polymath-eric**

### CODE REVIEW

File: src/api/procedures/utils.ts [hunk 1]
```diff
@@ -486,6 +455,42 @@ export async function assertTransferAssetOwnershipAuthorizationValid(
...
+ *
+ * Asserts valid add multisig signer authorization
+ */
+export async function assertMultiSigSignerAuthorizationValid(
+  data: GenericAuthorizationData,
+  target: Signer,
+  context: Context
+): Promise<void> {
+  if (target instanceof Account) {
+    if (target.address === data.value) {
```
Comment (monitz87) [inline src/api/procedures/utils.ts:469]: destructure `target.address` and reuse it in the `keyToMultiSig` call

File: src/api/procedures/consumeAddMultiSigSignerAuthorization.ts [hunk 2]
```diff
@@ -102,24 +92,24 @@ export async function getAuthorization(
...
+  let permissions;
   if (target instanceof Account) {
     calledByTarget = currentAccount.address === target.address;
     hasRoles = calledByTarget;
-    transactions = [TxTags.multiSig.AcceptMultisigSignerAsKey];
+    // An account accepting multisig cannot be part of an Identity, so we cannot check for permissions
   } else {
     calledByTarget = !!identity?.isEqual(target);
     hasRoles = calledByTarget;
     transactions = [TxTags.multiSig.AcceptMultisigSignerAsIdentity];
```
Comment (monitz87) [inline src/api/procedures/consumeAddMultiSigSignerAuthorization.ts:103]: No need for `transactions` to be defined in the outer scope if it's only used here

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/consumeAddMultiSigSignerAuthorization.ts [hunk 3]
```diff
@@ -100,7 +98,7 @@ export async function getAuthorization(
   } else {
     calledByTarget = !!identity?.isEqual(target);
     hasRoles = calledByTarget;
-    transactions = [TxTags.multiSig.AcceptMultisigSignerAsIdentity];
+    const transactions = [TxTags.multiSig.AcceptMultisigSignerAsIdentity];
     permissions = { transactions };
```
Comment (monitz87) [inline src/api/procedures/consumeAddMultiSigSignerAuthorization.ts:102]:
```suggestion
    permissions = { transactions:  [TxTags.multiSig.AcceptMultisigSignerAsIdentity] };
```

I didn't realize it was being used later. This should get rid of the code smell

**Review by monitz87 — CHANGES_REQUESTED** [review]
There's a linting error

File: src/api/procedures/__tests__/utils.ts [hunk 4]
```diff
@@ -609,7 +609,7 @@ describe('authorization request validations', () => {
   });
 
   describe('assertAuthorizationRequestValid', () => {
-    test('should throw with an expired request', () => {
+    it('should throw with an expired request', async () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/utils.ts:612]: Is it necessary for these to be async?

File: src/api/procedures/__tests__/utils.ts [hunk 5]
```diff
@@ -843,8 +841,8 @@ describe('authorization request validations', () => {
     });
   });
 
-  describe('assertPortfolioCustodyAuthorizationValid', () => {
-    test('should not throw with a valid request', () => {
+  describe('PortfolioCustody', () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/utils.ts:844]: Why this change?

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/__tests__/utils.ts (hunk 4 above)
Comment (polymath-eric) [reply src/api/procedures/__tests__/utils.ts:612]: No, I stripped them out before, but got hit with a merge conflict that reintroduced them.

File: src/api/procedures/__tests__/utils.ts (hunk 5 above)
Comment (polymath-eric) [reply src/api/procedures/__tests__/utils.ts:844]: Merge conflict, it used to be bad but I fixed it in this PR, then I wasn't careful resolving the conflict with the it changes.

**Review by monitz87 — APPROVED** [review]
